### ISSUE TITLE
(Dev version) Uploads not resumed after aeroplane mode or reboot

### ISSUE THREAD

**Issue by detly** [nextcloud/android#1251]
### Actual behaviour
- After queuing up some uploads and toggling aeroplane mode, the upload queue is not resumed.

### Expected behaviour
- Uploads should resume when a connection is available.
 
### Steps to reproduce
1. Manually upload a large amount of files comprising a large amount of data (in my case about 50 files, photos and videos, 4.5GB in total).
2. Wait for a few files to upload (screenshot 01).
3. Enable aeroplane mode. The uploads will stop.
4. Disable aeroplane mode. The uploads will remain stopped (screenshot 02).
5. Wait a few minutes. Soft-reboot the phone.
6. Wait a few more minutes. Launch Nextcloud app. Uploads list show queued uploads with "app terminated" status (screenshot 03).
7. Open the "recent apps" list, swipe to close Nextcloud. Re-launch Nextcloud. *Now* the uploads resume (screenshot 04).

### Environment data
Android version: 7.1.2

Device model: HTC One M9

Stock or customized system: Customised (LineageOS `14.1-20170725-NIGHTLY-himaul`)

Nextcloud app version: `20170727`

Nextcloud server version: 12.0

#### Web server error log

Not attached because it didn't seem necessary. Tell me if you want me to attach it.

#### Nextcloud log (data/nextcloud.log)

Exported from app settings. Attached: [`NC Log 01.txt`](https://github.com/nextcloud/android/files/1188976/NC.Log.01.txt)

Screenshots:

![screenshot 01](https://user-images.githubusercontent.com/311239/28800925-7f617512-7692-11e7-92f9-5fe50638f4d3.png)
![screenshot 02](https://user-images.githubusercontent.com/311239/28800926-7f8b3456-7692-11e7-81c4-75146519e04b.png)
![screenshot 03](https://user-images.githubusercontent.com/311239/28800927-7fb69ea2-7692-11e7-8ad1-fbefeb043cba.png)
![screenshot 04](https://user-images.githubusercontent.com/311239/28800928-7fe75a38-7692-11e7-94e0-32dd3ad901d1.png)

**Comment by mario** [comment]
I will assume this is my special version :) It resumes either when you force close/open the app or every 15-20 minutes.

With newest APK it should detect network changes instantly and resume the uploads.

https://cloud.nextcloud.com/s/pAbCOxR71TEhGgR

**Comment by detly** [comment]
Hah! I waited six minutes each time. I should've waited 7½. I'll try the
new version.

On 1 Aug. 2017 8:31 am, "Mario Đanić" <notifications@github.com> wrote:

I will assume this is my special version :) It resumes either when you
force close/open the app or every 15-20 minutes.

With newest APK it should detect network changes instantly and resume the
uploads.

https://cloud.nextcloud.com/s/pAbCOxR71TEhGgR

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/nextcloud/android/issues/1251#issuecomment-319215200>,
or mute the thread

**Comment by mario** [comment]
@detly news? :)

**Comment by detly** [comment]
Well I've been trying it out for the last few days, and overall it works really well! In summary:

 * My photos and videos get uploaded without my intervention! 👍 
 * Uploads are nearly always reliably restarted! 👍 
 * Auto-uploads reliably stop when on mobile data! 👍 

Problems I've encountered:
  * I haven't observed the feature you mentioned, where it detects a wifi connection and resumes uploading immediately. It always takes a good few minutes (10-20) to resume uploading. For my use case (taking a billion photos and videos of the baby to sort out later when I have time and a good night's sleep), I don't really care how quickly the auto-upload kicks in. It can just run overnight.
 * At one point it was repeatedly failing to upload, with some error about not being able to connect to the server. But uploading files with Solid Explorer worked, so the server and my connection were fine. This continued over a couple of hours, even though it seemed to be resuming the uploads, it would hit the same error every time for multiple files. I haven't been able to reproduce this situation though, and I don't think I have logs for it, so feel free to forget about it until I can get more information on it.
 * Progress indicators on the lock screen/status bar notifications are not consistent with those in the app upload queue. The lock screen notification will show a partially completed progress bar while the in-app upload list shows an empty progress bar (ie. 0%). Not really a big deal, but it means I can't be sure whether the uploads resume from where they left off (which would be a good thing for eg. long videos) or whether they always restart from 0%.

In short, I am super happy that unattended uploads now work so reliably and without me needing to constantly tap tap tap away, and I'm extremely grateful for the work that's gone into this.

**Comment by mario** [comment]
@detly thanks for the update. Let me comment on your observed problems:

- instant wifi change detection was added later, so you probably don't have the newest APK :)
- sounds strange.
- this is a tricky one. Resume from where it left off is still strange, and I'm unsure if it's working as perfect as it could. But yes, I will look into this later on.

Closing this for now - thank you for your awesome feedback!
